### PR TITLE
Preserve field order when updating kustomization.yaml file

### LINKED ISSUE #502: Retain or sort YAML ordering of images component when using Git write-back method
**Is your feature request related to a problem? Please describe.**

When the Image Updater updates kustomiztion.yaml file it does not sort the `image` component for new entries (nor retain existing formatting).

```
- name: ...
  digest: ...
```

vs (say via the use of `kustomize`)
```
  - digest: ...
    name: ...
```

**Describe the solution you'd like**
I would like the ordering to be sorted, or to retain the existing formatting.

**Additional context**
This is useful for us because we perform `git diff` comparisons (golden-tests) as part of our CI on our rendered manifests. Because argocd-image-updater changes the ordering, we frequently hit failed pipelines.

### PR DESCRIPTION
**Pull request by jtdoepke**
This commit causes argocd-image-updater to preserve the order of fields in the `kustomization.yaml` file when updating the image(s). Doing so will minimize git diffs in the case that other tools also edit the `kustomization.yaml` file.

Fixes #502

### CODE REVIEW

**Comment by jtdoepke** [comment]
The `build_image` job failed, but I'm pretty sure that's not due to my changes.

File: pkg/argocd/git_test.go [hunk 1]
```diff
@@ -215,3 +216,74 @@ images:
...
+
+	filter, err := imagesFilter(v1alpha1.KustomizeImages{"foo@sha23456"})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name        string
+		content     []byte
+		wantContent []byte
```
Comment (chengfang) [inline pkg/argocd/git_test.go:245]: can we keep `content` and `wantContent` as string to avoid multiple conversions?

**Comment by chengfang** [comment]
@jtdoepke can you also fix the DCO failure with sign-off?

**Review by chengfang — APPROVED** [review]

**Comment by chengfang** [comment]
@jtdoepke thanks!

**Pull request merged**